libobs/obs.c

```diff
@@ -1469,7 +1469,7 @@ void obs_enum_sources(bool (*enum_proc)(void *, obs_source_t *), void *param)
 {
 	obs_source_t *source;
 
-	pthread_mutex_lock(&obs->data.sources_mutex);
+	pthread_rwlock_rdlock(&obs->data.sources_rwlock);
```

Comment:
Given my previous statement about recursive mutexes, this would either have to be an rwlock, or it would need to make an incremented copy of the sources array, release the lock, then perform the enumeration on the copy.